src/storage/hummock_sdk/src/compaction_group/hummock_version_ext.rs

```diff
@@ -777,58 +797,70 @@ pub fn new_sub_level(
 ) {
-    if sub_level_idx < l0.sub_levels.len() {
-        insert_table_infos.iter().for_each(|sst| {
-            l0.sub_levels[sub_level_idx].total_file_size += sst.file_size;
-            l0.sub_levels[sub_level_idx].uncompressed_file_size += sst.uncompressed_file_size;
-            l0.total_file_size += sst.file_size;
-            l0.uncompressed_file_size += sst.uncompressed_file_size;
-        });
+    insert_table_infos.iter().for_each(|sst| {
+        l0.sub_levels[sub_level_idx].total_file_size += sst.file_size;
+        l0.sub_levels[sub_level_idx].uncompressed_file_size += sst.uncompressed_file_size;
+        l0.total_file_size += sst.file_size;
+        l0.uncompressed_file_size += sst.uncompressed_file_size;
+    });
+    l0.sub_levels[sub_level_idx]
+        .table_infos
+        .extend(insert_table_infos);
+    if l0.sub_levels[sub_level_idx].level_type == LevelType::Nonoverlapping as i32 {
         l0.sub_levels[sub_level_idx]
             .table_infos
-            .extend(insert_table_infos);
-        if l0.sub_levels[sub_level_idx].level_type == LevelType::Nonoverlapping as i32 {
-            l0.sub_levels[sub_level_idx]
-                .table_infos
-                .sort_by(|sst1, sst2| {
-                    let a = sst1.key_range.as_ref().unwrap();
-                    let b = sst2.key_range.as_ref().unwrap();
-                    a.compare(b)
-                });
-        }
-        return;
+            .sort_by(|sst1, sst2| {
+                let a = sst1.key_range.as_ref().unwrap();
+                let b = sst2.key_range.as_ref().unwrap();
+                a.compare(b)
+            });
     }
-    add_new_sub_level(l0, insert_sub_level_id, level_type, insert_table_infos);
 }
 
-pub fn add_new_sub_level(
+/// `None` value of `sub_level_insert_hint` means append.
+pub fn insert_new_sub_level(
     l0: &mut OverlappingLevel,
     insert_sub_level_id: u64,
     level_type: LevelType,
     insert_table_infos: Vec<SstableInfo>,
+    sub_level_insert_hint: Option<usize>,
 ) {
     if insert_sub_level_id == u64::MAX {
         return;
     }
-    if let Some(newest_level) = l0.sub_levels.last() {
-        assert!(
-            newest_level.sub_level_id < insert_sub_level_id,
-            "inserted new level is not the newest: prev newest: {}, insert: {}. L0: {:?}",
-            newest_level.sub_level_id,
-            insert_sub_level_id,
-            l0,
-        );
+    let insert_pos = if let Some(insert_pos) = sub_level_insert_hint {
+        insert_pos
+    } else {
+        if let Some(newest_level) = l0.sub_levels.last() {
+            assert!(
+                newest_level.sub_level_id < insert_sub_level_id,
+                "inserted new level is not the newest: prev newest: {}, insert: {}. L0: {:?}",
+                newest_level.sub_level_id,
+                insert_sub_level_id,
+                l0,
+            );
+        }
+        l0.sub_levels.len()
+    };
+    #[cfg(debug_assertions)]
```

Comment:
Does this operation happen very frequently? If not I think we can also enable the assertion.

Reply:
> Does this operation happen very frequently? If not I think we can also enable the assertion.

cc @Little-Wallace

Reply:
I think it is infrequent, the granularity of `move table` is minute level